cdm/core/src/main/java/ucar/nc2/ft2/coverage/writer/CoverageAsPoint.java

```diff
@@ -64,92 +84,333 @@
       this.cov = cov;
       try {
         this.array = cov.readData(subset);
-        if (debug)
+        if (debug) {
           System.out.printf(" Coverage %s data shape = %s%n", cov.getName(),
               Arrays.toString(array.getData().getShape()));
+        }
       } catch (InvalidRangeException e) {
         e.printStackTrace();
       }
     }
   }
 
   public CoverageAsPoint(CoverageCollection gcd, List<String> varNames, SubsetParams subset) throws IOException {
-    this.gcd = gcd;
     this.subset = subset;
 
-    latLonPoint = (LatLonPoint) subset.get(SubsetParams.latlonPoint);
-    if (latLonPoint == null)
+    this.latLonPoint = (LatLonPoint) subset.get(SubsetParams.latlonPoint);
+    if (latLonPoint == null) {
       throw new IllegalArgumentException("No latlon point");
+    }
 
-    varData = new ArrayList<>(varNames.size());
-    for (String varName : varNames) {
-      Coverage cov = gcd.findCoverage(varName); // LOOK we should read all at once if possible
-      if (cov != null) {
-        varData.add(new VarData(cov));
-
-        if (dateUnit == null) { // assume all have the same time unit, just use the first one
-          CoverageCoordSys csys = cov.getCoordSys();
-          CoverageCoordAxis timeAxis = csys.getTimeAxis();
-          this.dateUnit = timeAxis.getCalendarDateUnit();
-        }
-        if (nearestLatLonPoint == null) {
-          CoverageCoordSys subsetCs = varData.get(0).array.getCoordSysForData();
-          // single point subset, so only one lat/lon to grab, and this will be the lat/lon
-          // closest to the one requested in the subset
-          nearestLatLonPoint = subsetCs.getHorizCoordSys().getLatLon(0, 0);
-        }
+    // build var list, grouped by shared axes
+    varGroups = new ArrayList<>();
+    for (CoordSysSet css : gcd.getCoverageSets()) {
+      List<Coverage> wantCovs =
+          css.getCoverages().stream().filter(cov -> varNames.contains(cov.getName())).collect(Collectors.toList());
+      if (!wantCovs.isEmpty()) {
+        varGroups.add(new VarGroup(css.getCoordSys().getName(), wantCovs));
       }
     }
   }
 
   public FeatureDatasetPoint asFeatureDatasetPoint() {
-    // for the moment, assume a single station, no vert coord, single lat/lon
-    return new CoverageAsFeatureDatasetPoint(FeatureType.STATION);
+    return varGroups.size() == 0 ? null : new CoverageAsFeatureDatasetPoint();
```

Comment:
Maybe a tiny bit nicer:
```suggestion
    return varGroups.isEmpty() ? null : new CoverageAsFeatureDatasetPoint();
```